src/OrbitGl/GpuSubmissionTrack.h

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_GL_GPU_SUBMISSION_TRACK_H_
+#define ORBIT_GL_GPU_SUBMISSION_TRACK_H_
+
+#include <stdint.h>
+
+#include <memory>
+#include <string>
+#include <string_view>
+
+#include "CallstackThreadBar.h"
+#include "CoreMath.h"
+#include "GpuDebugMarkerTrack.h"
+#include "PickingManager.h"
+#include "StringManager.h"
+#include "TextBox.h"
+#include "TimerTrack.h"
+#include "Track.h"
+#include "capture_data.pb.h"
+
+class OrbitApp;
+class TextRenderer;
+
+// This track displays the "vkQueueSubmit"s including its hardware execution times and -- if
+// available -- command buffer timings on a certain command queue. In order to visually separate
+// different submissions that would overlap, they are displayed on a different depth with a thin gap
+// between them.
+// This track is meant to be used as a subtrack of `GpuTrack`.
+class GpuSubmissionTrack : public TimerTrack {
+ public:
+  explicit GpuSubmissionTrack(CaptureViewElement* parent, TimeGraph* time_graph,
+                              TimeGraphLayout* layout, uint64_t timeline_hash, OrbitApp* app,
+                              const CaptureData* capture_data, uint32_t intent);
+  ~GpuSubmissionTrack() override = default;
+  [[nodiscard]] Type GetType() const override { return Type::kGpuTrack; }
```

Comment:
Should this have its own type (kGpuSubTrack)?

Reply:
Honestly, there simply shouldn't be any value called type. The type was used by the track manager/timegraph to separate tracks and to pass the "correct" timers to them.  This has however already changed, and should be done by the actual runtime type of the track. It is currently only used by the "AddFrameTrack" functions to insert the track at the desired position.

Anyways, sorry for the rant. There is no need for subtracks to have a type, I am going to use the already existing `kUnknown`, in both subtracks.

Reply:
Add a comment explaining the choice of kUnknown.

Reply:
Done.